notebooks/Numba_053_CUDA_Release_Demo.ipynb

```diff
@@ -0,0 +1,1285 @@
+    "        g.sync()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "engaged-temple",
+   "metadata": {},
+   "source": [
+    "We'll create an empty matrix for the kernel to work on, and determine an appropriate block size:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "purple-burns",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Empty input data\n",
+    "A = np.zeros((1024, 1024), dtype=np.int32)\n",
+    "# A somewhat arbitrary choice (one warp), but generally smaller block sizes\n",
+    "# allow more blocks to be launched (noting that other limitations on\n",
+    "# occupancy apply such as shared memory size)\n",
+    "blockdim = 32\n",
+    "griddim = A.shape[1] // blockdim"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "motivated-tyler",
+   "metadata": {},
+   "source": [
+    "Now we can launch the kernel and print the result:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "painted-compilation",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[[   0    0    0 ...    0    0    0]\n",
+      " [   1    1    1 ...    1    1    1]\n",
+      " [   2    2    2 ...    2    2    2]\n",
+      " ...\n",
+      " [1021 1021 1021 ... 1021 1021 1021]\n",
+      " [1022 1022 1022 ... 1022 1022 1022]\n",
+      " [1023 1023 1023 ... 1023 1023 1023]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Kernel launch - this is implicitly a cooperative launch\n",
+    "sequential_rows[griddim, blockdim](A)\n",
+    "\n",
+    "# What do the results look like?\n",
+    "print(A)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "independent-midnight",
+   "metadata": {},
+   "source": [
+    "There is a more strict limit on the grid size for launching a kernel using a Grid Group, as all blocks must be able to be resident on the GPU concurrently - unlike a regular launch, it is not possible to wait for one block to fully execute before another block launches, as this would create the conditions for a deadlock.\n",
+    "\n",
+    "The maximum grid size for a cooperative kernel can be enquired with `max_cooperative_grid_blocks()` (and varies between GPU models, depending on the number of SMs and their available resources):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "fifth-fraction",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1152\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/gmarkall/numbadev/numba/numba/cuda/compiler.py:1110: NumbaDeprecationWarning: Use overloads instead of definition\n",
+      "  warn('Use overloads instead of definition',\n"
+     ]
+    }
+   ],
+   "source": [
+    "defn = sequential_rows.definition\n",
```

Comment:
Done - originally I didn't want to use `overloads` because it's not quite as elegant (but doesn't align with a longer-term plan to support multiple signatures in the CUDA target), but it's better not to present code that pops a deprecation warning.